src/tiled/mapeditor.cpp

```diff
@@ -314,6 +316,10 @@ void MapEditor::addDocument(Document *document)
         if (Layer *layer = layerAtGlobalIndex(mapDocument->map(), layerIndex))
             mapDocument->setCurrentLayer(layer);
     }
+
+    // Update all templates instances in this MapDocument when a template is edited
+    // Maybe it can be optimized
```

Comment:
That would be because the changing object is a clone of the template object, so indeed the template object needs to be synchronized to its clone each time before the instances are synced with the template object. But sure, we can wait with this change and making it would definitely also require optimization to how the instances are updated.